libraries/react/src/basic-tests.js

```diff
@@ -72,30 +97,57 @@ describe("basic support", function() {
 
     it("can display a Custom Element with children in a Shadow Root", function() {
       this.weight = 3;
-      let root = ReactDOM.render(<ComponentWithChildren />, scratch);
+      let root;
+      render(
+        <ComponentWithChildren
+          ref={(current) => {
+            root = current;
+          }}
+        />
+      );
       let wc = root.wc;
       expectHasChildren(wc);
     });
 
     it("can display a Custom Element with children in a Shadow Root and pass in Light DOM children", async function() {
       this.weight = 3;
-      let root = ReactDOM.render(<ComponentWithChildrenRerender />, scratch);
+      let root;
+      render(
+        <ComponentWithChildrenRerender
+          ref={(current) => {
+            root = current;
+          }}
+        />
+      );
       let wc = root.wc;
-      await Promise.resolve();
+      await act(async () => {
```

Comment:
Is it necessary to wrap await Promise.resolve() in act?

Reply:
Yes. Any state update needs to be wrapped in `act` in React 18. `ComponentWithChildrenRerender` schedules a state update in a microtask i.e. `Promise.resolve().then(() => this.setState())` so we need to wrap that update in `act`